### PR TITLE
Support for UTF-8 (see issue #16)

### PR DESCRIPTION
**Pull request by denis-gz**

### CODE REVIEW

**Comment by stachenov** [comment]
I haven't looked at it yet, but “emit always” is definitely not acceptable due to backward compatibility. It will break existing use cases where people just want to create legacy ZIP archives. Maybe an archive-wide flag, similar to setDataDescriptorWritingEnabled will do?

File: quazip/quazipfile.cpp [hunk 1]
```diff
@@ -340,10 +340,10 @@ bool QuaZipFile::open(OpenMode mode, const QuaZipNewInfo& info,
     else
         zipClearFlags(p->zip->getZipFile(), ZIP_WRITE_DATA_DESCRIPTOR);
     p->setZipError(zipOpenNewFileInZip3_64(p->zip->getZipFile(),
-          p->zip->getFileNameCodec()->fromUnicode(info.name).constData(), &info_z,
+          qUtf8Printable(info.name), &info_z,
```
Comment (stachenov) [inline quazip/quazipfile.cpp]: This won't do. `qUtf8Printable` was introduced in Qt 5.4. QuaZIP supports backward compatibility with Qt 4.x. Better replace with `toUtf8().constData()` here and in other places too.

Comment (denis-gz) [reply quazip/quazipfile.cpp]: Ok it's easy.

**Comment by denis-gz** [comment]
> I haven't looked at it yet, but “emit always” is definitely not acceptable due to backward compatibility. It will break existing use cases where people just want to create legacy ZIP archives. Maybe an archive-wide flag, similar to setDataDescriptorWritingEnabled will do?

This is going to complicate things a little, but probably it's the right way to go.

**Comment by denis-gz** [comment]
Changes made as you requested.

**Comment by stachenov** [comment]
Looks nice, tested and merged successfully.

**Pull request merged**